UKHomeOffice/cop-ui
# COP-8543 Add targets list & info page

## Notes

Ended up needing to refactor the pagination & targetInfo page as part of this to get the ability for any user to view the targets, and for claim to happen on the 'record outcome' click rather on a claim button, hence the large PR.

## To Test

One: normal form
- go to your team targets
- claim any task that is not a Targeting Task
- > you should be taken to the form for that task

Two: normal unclaimed
- go to your team targets
- click on a task that you have not claimed
- > you should see the 'you must claim message

Three: target from group list
- go to your team targets
- click on a Target Task (not the claim button)
- > you should see the target info page

Four: target list
- go to Targets from the dashboard
- > you should see a list of targets
- > NOTE: targets are not currently sorted due to a BPMN bug that will be fixed in COP-8659

Five: target from target list
- go to Targets from the dashboard
- click on a Target Task
- > you should see a the target info page

Six: complete a target
- go to Targets from the dashboard
- click on a Target Task
- click on 'record outcome'
- > you should see the record outcome form
- complete the form
- > you should be able to complete the form

Seven: see who has claimed a target
- go to your team tasks
- find a target task that is claimed by someone else
- click on that task (not the claim button)
- at the bottom of the target info page you should see who has claimed that task

Eight: complete a target claimed by someone else
- go to your team tasks
- find a target task that is claimed by someone else
- click on that task (not the claim button)
- click on 'record outcome'
- > you should see the record outcome form
- complete the form
- > you should be able to complete the form

Nine: pagination

- Go to cop-ui Targets (/target)
- if there are more than 20 tasks you should have pagination options [1, [x], next, last >>]
- click on any link, it should update the tasks to show ones for that page
- if you are on a page other than 1 you should have the options [<<first, previous, 1, [x], next, last>>]
- if you are on the last page you should not have next/last as options
- on the right hand side you should see the number of the tasks being shown
-
